wandb/jupyter.py

```diff
@@ -36,6 +39,63 @@ def wandb(self, line, cell=None):
         if cell is not None:
             get_ipython().run_cell(cell)
 
+def attempt_colab_login(app_url):
+    """This renders an iframe to wandb in the hopes it posts back an api key"""
+    from google.colab import output
+    from google.colab._message import MessageError
+    from IPython import display
+
+    display.display(display.Javascript('''
+        window._wandbApiKey = new Promise((resolve, reject) => {
+            function loadScript(url) {
+            return new Promise(function(resolve, reject) {
+                let newScript = document.createElement("script");
+                newScript.onerror = reject;
+                newScript.onload = resolve;
+                document.body.appendChild(newScript);
+                newScript.src = url;
+            });
+            }
+            loadScript("https://cdn.jsdelivr.net/npm/postmate/build/postmate.min.js").then(() => {
+            const iframe = document.createElement('iframe')
+            iframe.style.cssText = "width:0;height:0;border:none"
+            document.body.appendChild(iframe)
+            const handshake = new Postmate({
+                container: iframe,
+                url: '%s/authorize'
+            });
+            const timeout = setTimeout(() => reject("Couldn't auto authenticate"), 5000)
+            handshake.then(function(child) {
+                child.on('authorize', data => {
+                    clearTimeout(timeout)
+                    resolve(data)
+                });
+            });
+            })
+        });
+    ''' % app_url.replace("http:", "https:")))
+    try:
+        return output.eval_js('_wandbApiKey')
+    except MessageError:
+        return None
+
+def notebook_metadata():
+    """Attempts to query jupyter for the path and name of the notebook file"""
+    try:
+        import ipykernel
+        from notebook.notebookapp import list_running_servers
+        kernel_id = re.search('kernel-(.*).json', ipykernel.connect.get_connection_file()).group(1)
+    except Exception:
+        return {}
+    for s in list_running_servers():
+        try:
+            res = requests.get(urljoin(s['url'], 'api/sessions'), params={'token': s.get('token', '')}).json()
+        except (requests.RequestException, ValueError):
```

Comment:
What sort of thing would cause this? Not having a token? Is this a pretty normal situation?

Reply:
I just added logging, this tries to talk to the local jupyter kernel which I assume could throw exceptions due to network issues.  It should happen very rarely if ever.